### PR TITLE
ifdef windows specific code

### PR DESCRIPTION
**Pull request by karthiknadig**

### CODE REVIEW

File: src/Microsoft.R.Host.vcxproj [hunk 1]
```diff
@@ -103,7 +103,7 @@
       </PrecompiledHeader>
       <WarningLevel>Level3</WarningLevel>
       <Optimization>Disabled</Optimization>
-      <PreprocessorDefinitions>WIN32;_DEBUG;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>TRACE_JSON;WIN32;_DEBUG;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
Comment (int19h) [inline src/Microsoft.R.Host.vcxproj]: `WIN32` is defined without the underscore here, but you use `_WIN32` below - is that intentional?

File: src/transport.cpp [hunk 2]
```diff
@@ -98,9 +98,9 @@ namespace rhost {
             setmode(fileno(stdout), _O_BINARY);
 
             // Duplicate and stash away handles for original stdin & stdout.
-            input = fdopen(dup(fileno(stdin)), "rb");
+            input = fdopen(_dup(fileno(stdin)), "rb");
```
Comment (int19h) [inline src/transport.cpp]: It needs to be `dup` to be cross-plat (it's a POSIX function).

File: src/Microsoft.R.Host.vcxproj (hunk 1 above)
Comment (karthiknadig) [inline src/Microsoft.R.Host.vcxproj]: VS C++ console project template has that by default for 32 bit builds.

File: src/transport.cpp (hunk 2 above)
Comment (karthiknadig) [inline src/transport.cpp]: I see, I will change that.

**Comment by int19h** [comment]
👍

**Comment by MikhailArkhipov** [comment]
Master is RC2 at the moment, shall we take this or delay?

**Comment by int19h** [comment]
It's being merged into a separate branch, not master.

**Comment by MikhailArkhipov** [comment]
OK

**Pull request merged**